contracts/core/Allo.sol

```diff
@@ -77,44 +76,66 @@ contract Allo is IAllo, Native, Transfer, Initializable, Ownable, AccessControl
         external
         reinitializer(1)
     {
+        // Initialize the owner using Solady ownable library
         _initializeOwner(msg.sender);
 
+        // Set the address of the registry
         _updateRegistry(_registry);
+
+        // Set the address of the treasury
         _updateTreasury(_treasury);
+
+        // Set the fee percentage
         _updatePercentFee(_percentFee);
+
+        // Set the base fee
         _updateBaseFee(_baseFee);
     }
 
-    /// ====================================
-    /// =========== Modifier ===============
-    /// ====================================
+    // ====================================
+    // =========== Modifier ===============
+    // ====================================
+
+    /// Both modifiers below are using OpenZeppelin's AccessControl.sol with custom roles under the hood
 
+    /// @notice Reverts UNAUTHORIZED() if the caller is not a pool manager
+    /// @param _poolId The pool id
     modifier onlyPoolManager(uint256 _poolId) {
         if (!_isPoolManager(_poolId, msg.sender)) {
             revert UNAUTHORIZED();
         }
         _;
     }
 
+    /// @notice Reverts UNAUTHORIZED() if the caller is not a pool admin
+    /// @param _poolId The pool id
     modifier onlyPoolAdmin(uint256 _poolId) {
         if (!_isPoolAdmin(_poolId, msg.sender)) {
             revert UNAUTHORIZED();
         }
         _;
     }
 
-    /// ====================================
-    /// ==== External/Public Functions =====
-    /// ====================================
+    //  ====================================
+    //  ==== External/Public Functions =====
+    //  ====================================
 
     /// @notice Creates a new pool (with custom strategy)
-    /// @param _profileId The profileId of the pool
-    /// @param _strategy The address of strategy
+    /// @dev Must be a member or owner of a profile to create a pool with or without a custom strategy
+    ///
+    /// Requirements: The strategy address passed must not be a cloneable strategy
+    ///               The strategy address passed must not be the zero address
+    ///               The caller must be a member or owner of the profile
```

Comment:
Suggestion:
The caller must be a member or owner of the `related registry` profile

Reply:
the comment from `createPool` is nice:
`/// Requirements: 'msg.sender' must be owner or member of the profile id passed as '_profileId'`